core/src/main/java/org/apache/cxf/io/ReaderInputStream.java

```diff
@@ -120,9 +119,9 @@ public ReaderInputStream(Reader reader, CharsetEncoder encoder, int bufferSize)
         this.reader = reader;
         this.encoder = encoder;
         this.encoderIn = CharBuffer.allocate(bufferSize);
-        ((Buffer)this.encoderIn).flip();
+        (this.encoderIn).flip();
```

Comment:
flip() do nothing for new allocated buffer

Reply:
@amarkevich Done with thanks . Good observation

Reply:
it affect encoderIn buffer too

Reply:
@amarkevich  . Apology I missed  it Good you have spotted it . Done with thanks